.storybook/preview.tsx

```diff
@@ -162,18 +162,24 @@ const withThemeSwitcher: Decorator = (
 };
 
 /**
- * Wraps a story with `<div dir="rtl">` so it is shown in rtl mode.
+ * Sets the dir attribute on document.documentElement and wraps story with dir div.
+ * This ensures portaled content (like modals) can detect the direction.
  */
 const withLanguageDirection: Decorator = (Story, context) => {
-    if (context.globals.direction === "rtl") {
-        return (
-            <div dir="rtl">
-                <Story />
-            </div>
-        );
-    } else {
-        return <Story />;
-    }
+    React.useEffect(() => {
+        if (context.globals.direction === "rtl") {
+            document.documentElement.setAttribute("dir", "rtl");
+        } else {
+            document.documentElement.setAttribute("dir", "ltr");
+        }
+
+        // Cleanup on unmount
+        return () => {
+            document.documentElement.removeAttribute("dir");
+        };
+    }, [context.globals.direction]);
+
+    return <Story />;
```

Comment:
We no longer need to wrap the Story with a `<div dir="rtl">`, but we'll need stories to re-render when the direction changes! I think we can achieve that by setting a `key` on `<Story />` to `context.globals.direction` so it re-renders when the value changes!

While testing `Tabs`, I noticed the active styling wasn't properly updating like it used to! (Tabs Story ([before](https://khan.github.io/wonder-blocks/?path=/story/packages-tabs-tabs--default&globals=direction:ltr)/[after](https://5e1bf4b385e3fb0020b7073c-jcopircsoa.chromatic.com/?path=/story/packages-tabs-tabs--default&globals=direction:ltr))). The issue can be reproduced by changing the writing direction using the control! 

https://github.com/user-attachments/assets/6cf3752a-a1d3-4156-a356-6a4ce380f3ba

Reply:
Ah, good find! I'm going to punt on this change until later and restore the wrapper DIV with the document attribute also. The key isn't fixing the Tabs style issue for me, even with a dynamic `dir` attribute on that wrapper element. 

In debugging Tabs I also found the Popover CloseButton isn't updating with RTL. It wasn't obvious before, because it gets injected outside of the Story like Modals do! I filed an issue to split that work off: https://khanacademy.atlassian.net/browse/WB-2069